### DISCUSSION TITLE
Feature Request: Add support for aggregate ordering for function expressions

### DISCUSSION THREAD

**Discussion by Tmonster** [duckdb/duckdb#6914] [Ideas]
Currently the relational api has functionality for functions like "sum", "avg", "count()" etc. However, functions with an ordering attribute cannot be expressed using the relational api. Take the following SQL statement for example

```SQL
SELECT STRING_AGG(x,',' ORDER BY x ASC), STRING_AGG(x,y ORDER BY x ASC) FROM strings
```

Would be nice if this also eventually works for window statements, so something like the following
```SQL
SELECT STRING_AGG(x,',' ORDER BY x ASC), STRING_AGG(x,y ORDER BY x ASC) OVER (PARTITION BY part) FROM string
```

**Comment by hawkfish** [comment]
This is what [Adrian's paper](https://dl.acm.org/doi/pdf/10.1145/3514221.3526184) was all about and its on my radar. But I think I need to finish a bunch of refactoring around state management first.

**Comment by hawkfish** [comment]
Implemented.
